src/components/checkbox/checkbox.html

```diff
@@ -16,7 +30,7 @@
       <div class="md-checkbox-mixedmark"></div>
     </div>
   </div>
-  <label [id]="labelId">
+  <span class="md-checkbox-label">
```

Comment:
No - the outer wrapper is a label which automatically handles this properly since this relies on native behavior.